crates/catalog/glue/src/catalog.rs

```diff
@@ -395,10 +395,7 @@ impl Catalog for GlueCatalog {
             .metadata;
         let metadata_location = create_metadata_location(&location, 0)?;
 
-        self.file_io
-            .new_output(&metadata_location)?
-            .write(serde_json::to_vec(&metadata)?.into())
-            .await?;
+        TableMetadata::write(&self.file_io, &metadata, &metadata_location).await?;
```

Comment:
I also found `MetadataLocation` proposed [here](https://github.com/apache/iceberg-rust/pull/1405/files#diff-202adb538eb8bce001ab1109cd54b3fc4a607cbee4171aa6cb3f67d92718c453R330) quite interesting: 

Would love to hear more thoughts about it

Reply:
I changed it to 
```
fn write(&self, impl AsRef<str>)
```
because `file_io.new_input/new_output` takes `AsRef<str>`